### ISSUE TITLE
[Containers] : The $dotPageContent object is available only after refresh

### ISSUE THREAD

**Issue by jcastro-dotcms** [dotCMS/core#21286]
Reported via Support ticket: [105657](https://dotcms.zendesk.com/agent/tickets/105657)
Reproducible as of dotCMS 5.3.8.6.1, up to latest master (21.11)

Because of a code reuse approach used by the customer, the Container they use will render the Contents in it based on the subtype of the Page Asset used for the HTML Page. So, they need to get the Velocity Variable Name of the Content Type that is used by the HTML Page that renders the Container in order to use the appropriate HTML code for the Contents in it, like this:
```
$dotPageContent.structure.velocityVarName
```
However, this `$dotPageContent` object is NOT fully available when you add a new or existing Content to the Container. You need to refresh the browser's tab or switch to PREVIEW and back to EDIT Mode in order to see the Content displayed correctly. I reproduced this under supervision of @freddyucv .

**To Reproduce**
1. Create a test Container named `Test Container`. Set **Max Contents** to 5, select the `Rich Text` type, and add the following code:
```html
<h1>My Test Container</h1>
<p>
    $!{title}
</p>
#if ($UtilMethods.isSet($dotPageContent.structure.velocityVarName))
    <h3>dotPageContent object has the expected values</h4>
#else
    <h3>dotPageContent is NOT fully available...</h3>
#end
<hr/>
<h5>Contents of the dotPageContent object:</h5>
<div>
    $dotPageContent
</div>
``` 
2. Create an Advanced Template named `My Template`, and add the previous Container. For example:
```html
<html>
    <head>
        <title>My Test Template</title>
    </head>
    <body>
        ## Container: test container
        ## This is autogenerated code that cannot be changed
        #parseContainer('b678163d35890349aeb998660f3ac58a','1637345340039')
    </body>
</html>
```
3. Create a new HTML Page named `my-test-page` using the Template created previously, and go to the Edit Mode.
4. Click the **Add > Content** button in the Container, and select any piece of content of type `Rich Text`.
5. The output of the page is the following, which is NOT correct. For example:
```
My Test Container
Adventure travel done right

dotPageContent is NOT fully available...
Contents of the dotPageContent object:
true
```
Notice how the `$dotPageContent` is printed as `true` instead of the actual object's value. Given that the `$dotPageContent.structure.velocityVarName` is not defined, we're seeing the `dotPageContent is NOT fully available` message.
6. Go to **PREVIEW** and go back to **EDIT**. Notice how the `$dotPageContent` variable is being printed correctly:
```
My Test Container
Adventure travel done right

dotPageContent object has the expected values
Contents of the dotPageContent object:
com.dotcms.rendering.velocity.viewtools.content.ContentMap@37797fa8[content=com.dotmarketing.portlets.contentlet.model.Contentlet@2c2b8c42[map={owner=dotcms.org.1, template=6f2d5ce3bf26e04af58ce5b682432068, identifier=2d76186887476b5f0ce5a422db1aab50, nullProperties=[seoDescriptionCount, wfActionAssign, seoTitle, httpsreq,
...
...
...
```
The Velocity objects are showing up as expected. That's why we can see the `dotPageContent object has the expected values` message now.

**Expected behavior**
dotCMS should be able to access all the objects in the Velocity context even when a piece of Content is added, without having to refresh the page.

**Comment by dsilvam** [comment]
Passed Internal QA: $dotPageContent available in three modes since the first time. 

https://gist.github.com/dsilvam/ab9a7a3a200860162f514f8726365355

**Comment by victoralfaro-dotcms** [comment]
`$dotPageContent` variable is displayed correctly.

https://user-images.githubusercontent.com/64026161/146469217-3a26c9e3-13f3-457d-8b1a-480fcc88b886.mov

**Comment by bryanboza** [comment]
Fixed, tested on release-22.01 // Docker // FF
